merky/test/exclude_annotation_nesting_test.py

```diff
@@ -0,0 +1,153 @@
+
+
+
+
+def test_simple_unannotated_list():
+    # It always does the top level guy regardless of annotation.
+    t = merky.ExcludeAnnotationTransformer()
+    r = t.transform(BASIC_SEQ)
+    tools.assert_equal((BASIC_SEQ_HASH, list(BASIC_SEQ)), next(r))
+    tools.assert_raises(StopIteration, next, r)
+
+
+def test_nested_unannotated_list():
+    # Top level always gets done, but this time, the nested guy wil be too.
+    t = merky.ExcludeAnnotationTransformer()
+    r = t.transform(UNANNOTATED_NESTED_SEQ)
+    # We get the inner guy first, who is tokenized due to annotation
+    tools.assert_equal((BASIC_SEQ_HASH, list(BASIC_SEQ)), next(r))
+    # The outer guy is different this time, as the token replaced the nested list.
+    # SHA1 of json of [words.ANGSTROM, BASIC_SEQ_HASH, 213, -315]
+    tools.assert_equal(("3e2f759672d412ed84b7359e2613c94cfee250f2", [words.ANGSTROM, BASIC_SEQ_HASH, 213, -315]),
+                       next(r))
+    tools.assert_raises(StopIteration, next, r)
+
+
+def test_nested_annotated_list():
+    # And again, it always does the top level guy.  But since no members are annotated, it
+    # will explode out anything else.
+    t = merky.ExcludeAnnotationTransformer()
+    r = t.transform((words.ANGSTROM, merky.annotate(BASIC_SEQ), 213, -315))
+    tools.assert_equal((UNANNOTATED_NESTED_SEQ_HASH, [words.ANGSTROM, list(BASIC_SEQ), 213, -315]), next(r))
+    tools.assert_raises(StopIteration, next, r)
+
+
+def test_simple_unannotated_dict():
+    t = merky.ExcludeAnnotationTransformer()
+    r = t.transform(BASIC_DICT)
+    tools.assert_equal((BASIC_DICT_HASH, BASIC_DICT_ORDERED), next(r))
+    tools.assert_raises(StopIteration, next, r)
+
+
+def test_nested_unannotated_dict():
+    t = merky.ExcludeAnnotationTransformer()
+    r = t.transform(UNANNOTATED_NESTED_DICT)
+    # We get the inner dict first, who is tokenized due to lack of annotation
+    tools.assert_equal((BASIC_DICT_HASH, BASIC_DICT_ORDERED), next(r))
+    # We get the inner list second, who is tokenized due to lack of annotation
+    tools.assert_equal((BASIC_SEQ_HASH, list(BASIC_SEQ)), next(r))
+    # The outer guy is different this time, as tokens replace the nested dict and list
+    # SHA1 of json of [words.ANGSTROM, BASIC_SEQ_HASH, 213, -315]
+    tools.assert_equal((UNANNOTATED_NESTED_DICT_HASH, UNANNOTATED_NESTED_DICT_TOKENIZED),
+                       next(r))
+    tools.assert_raises(StopIteration, next, r)
+
+
+def test_mixed_nesting():
+    t = merky.ExcludeAnnotationTransformer()
+    x = {"a": merky.annotate([
+                    {"a": merky.annotate({"a": "A", "b": "B"}), "b": "B"},
+                    {"nothing": "special"},
+                ]),
+         "b": [
+                merky.annotate({"c": "C", "d": "D"}),
+                {"e": "E", "f": "F"},
+              ],
+        }
+    r = t.transform(x)
+    # Lowest level: x["a"][0]
+    tools.assert_equal(('9999c10f0f35c46a472e16ecec8d7cac6b0ac127',
```

Comment:
Ah, it's because the logic of just flipping the truth value in the tree functions isn't enough.  In this case, we have to interrupt the recursive descent, such that a higher-level annotation effectively cuts off further consideration of the innards.  That's harder to do than the change I showed yesterday.  Pooh.